sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -308,3 +308,94 @@ class AnyCombineFn(AccumulatorCombineFn):
 
 class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
+
+MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
+                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
+                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
+                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
+                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
+
+LONG_SIZE = 64
+
+def compare_to(x, y):
+  """return the sign bit of x-y"""
+  if x < y:
+    return 1
+  return 0
+
+def get_log10_round_to_floor(element):
+  number_of_leading_zeros = LONG_SIZE - element.bit_length()
+  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
+  return y - compare_to(element, math.pow(10, y))
+
+class DistributionAccumulator(object):
+  """Distribution Counter:
+  contains value distribution statistics and methods for incrementing
+  """
+  def __init__(self):
+    self.min = 0
+    self.max = 0
+    self.count = 0
+    self.sum = 0
+    self.sum_of_squares = 0
```

Comment:
If this is intended for variance computations, there are numerical stability issues here.

Reply:
Could you please explain more about your concerns here? Thanks for your help!

Reply:
```
In [1]: import numpy as np
In [2]: N = int(1e8)
In [3]: data = np.random.uniform(size=N)

In [4]: np.std(data)**2   # Actual variance, about 1/12
Out[4]: 0.08332694403630711

In [5]: 1/(N-1.0) * (np.sum(data**2) - np.sum(data)**2 / N)  # Sum of squares, not too bad
Out[5]: 0.083326944869713787

In [6]: data = np.random.uniform(size=N) + 1e4

In [6]: 1/(N-1.0) * (np.sum(data**2) - np.sum(data)**2 / N)  # OK, we've at least got a couple digits right
Out[14]: 0.083349540833495397

In [7]: data += 1e5

In [8]: 1/(N-1.0) * (np.sum(data**2) - np.sum(data)**2 / N)  # Variance should be the same, now we're pretty far off
Out[8]: 0.077086720770867201

In [9]: data += 1e6

In [10]: 1/(N-1.0) * (np.sum(data**2) - np.sum(data)**2 / N)  # Woah, this square is not even positive
Out[10]: -2.6093158660931586

In [11]: data += 1e7

In [12]: 1/(N-1.0) * (np.sum(data**2) - np.sum(data)**2 / N) # It's pretty much garbage from here on out...
Out[12]: 2138.2771721027716
```

Reply:
Deleted sum_of_squares